src/adaptors/amqp/container.c

```diff
@@ -645,6 +627,7 @@ void qd_container_handle_event(qd_container_t *container, pn_event_t *event,
         pn_link = pn_event_link(event);
         if (pn_link_state(pn_link) == (PN_LOCAL_CLOSED | PN_REMOTE_CLOSED)) {
             add_link_to_free_list(&qd_conn->free_link_list, pn_link);
+            qd_link_free((qd_link_t *) pn_link_get_context(pn_link));
```

Comment:
Instead of calling `qd_link_free()` once in the handler to `PN_LINK_REMOTE_CLOSE` and once in `PN_LINK_LOCAL_CLOSE`, can we call it just once in the `qd_conn_event_batch_complete()` function where we loop thru the `qd_conn->free_link_list` ?

In fact, I was wondering if we could have the `qd_conn_event_batch_complete()` function as the only place we call 'qd_link_free()' and 'pn_link_free()'